api/config/v1alpha1/accesslogging_types.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+type ProxyAccessLogging struct {
+	// Format defines the format of access logging.
+	Format ProxyAccessLoggingFormat `json:"format"`
+	// Sinks defines the sinks of access logging.
+	// +kubebuilder:validation:MinItems=1
+	Sinks []ProxyAccessLoggingSink `json:"sinks"`
+}
+
+type ProxyAccessLoggingFormatType string
+
+const (
+	// ProxyAccessLoggingFormatTypeText defines the text access logging format.
+	ProxyAccessLoggingFormatTypeText ProxyAccessLoggingFormatType = "Text"
+	// ProxyAccessLoggingFormatTypeJSON defines the JSON access logging format.
+	ProxyAccessLoggingFormatTypeJSON ProxyAccessLoggingFormatType = "JSON"
+	// TODO: support format type "mix" in the future.
+)
+
+type ProxyAccessLoggingFormat struct {
+	// Type defines the type of access logging format.
+	// +kubebuilder:validation:Enum=Text;JSON
+	Type ProxyAccessLoggingFormatType `json:"type,omitempty"`
+	// Text defines the text access logging format, following Envoy access logging formatting,
+	// empty value results in proxy's default access log format.
+	// It's required when the format type is "text".
+	// Envoy [command operators](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#command-operators) may be used in the format.
+	// The [format string documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#config-access-log-format-strings) provides more information.
+	// +optional
+	Text *string `json:"text,omitempty"`
+	// Fields is additional attributes that describe the specific event occurrence.
+	// Structured format for the envoy access logs. Envoy [command operators](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#command-operators)
+	// can be used as values for fields within the Struct.
+	// It's required when the format type is "json".
+	// +optional
+	Fields map[string]string `json:"fields,omitempty"`
```

Comment:
the top level field must be `Json` and inner field can be `fields`
here is a good example of union & unionDiscriminator tags https://github.com/envoyproxy/gateway/blob/a9a85281f03cada583f65dca61c111e7c36e10d9/api/v1alpha1/ratelimitfilter_types.go#L30

Reply:
I add them, but got nothing changed after running `make generate manifests`